File: tests/modules/src/module_b.rs
use crate::{FieldA, FieldB, StructA};
use tslink::tslink;

#[tslink(target = "./output/module_b.ts", module = "module_b")]
pub enum EntityA {
    One,
    Two,
    Three,
}

#[tslink(target = "./output/module_b.ts", module = "module_b")]
pub enum EntityB {
    One(String),
    Two((u32, u32)),
    Three(EntityA),
}

#[tslink(target = "./output/module_b.ts", module = "module_b")]
pub struct OtherStruct {
    pub a: EntityA,
    pub b: EntityB,
    pub c: StructA,
    pub d: FieldA,
    pub e: FieldB,
}

File: tests/modules/output/module_b.ts
export enum EntityA {
    One,
    Two,
    Three,
}
import { StructA } from "./module_a";
import { FieldA } from "./module_a";
import { FieldB } from "./module_a";
export interface OtherStruct {
    a: EntityA;
    b: EntityB;
    c: StructA;
    d: FieldA;
    e: FieldB;
}
export interface EntityB {
    One?: string;
    Two?: [number, number];
    Three?: EntityA;
}
